src/lib/qnio/utils.c

```diff
@@ -9,6 +9,117 @@
  */
 
 #include "defs.h"
+#include "qnio.h"
+
+int 
+is_secure()
+{
+    if (access(SECURE_IMPL, F_OK) != 0)
```

Comment:
IMO, having the qemu management layer exercise complete control over the block device characteristics and operations is a noble but non-trivial goal. There will always be certain aspects of the underlying device that need not be known to QEMU. 
E.g. 
1.  failover target information for vxhs vdisks. 
2. number of retries before failover
Are these something that you think needs to be driven through QEMU as well? That would mean that we need to add APIs like iio_set_failover_targets() API to pass this down from  qemu to the block device handler library. This BTW, was earlier being passed through libvirt but was changed to be handled locally by libvxhs after reviews from community.
Relying solely on qemu/libvirt mgmt interface to tune each and every one of the block device parameters would not be an efficient way to manage the block device.

Use of ssl/tls (if the user decides to enable it) is a global setting for VxHS block devices. Our use case doesn't demand this setting per instance. If VxHS determines this to be a useful setting on a per instance basis, then we will add the new API and the necessary management layer changes at a future date.

Also, it would really help if these requirements were published and maintained as guidelines for future drivers/contributions.